content/en/docs/plugins/developers_guide.md

```diff
@@ -0,0 +1,1081 @@
+
+At a high level, the SDK is on top of three fundamendal package with different levels of abstractions:
+
+1. Package `sdk` is a container for all the basic types, definitions, and helpers that are reused across all the SDK parts. 
+
+2. Package `sdk/symbols` contains prebuilt implementations for all the C symbols that plugins must export to be accepted by the framework. The prebuilt C symbols are divided in many subpackages, so that each of them can be imported individually to opt-in/opt-out each symbol. 
+
+3. Package `sdk/plugins` provide high-level definition and base types for writing extractor plugins and source plugins. This uses `sdk/symbols` internally and takes care of importing all the prebuilt C symbols required by extractor and source plugins respectively. This is the main entrypoint for developers to write plugins in Go.
+
+Two additional packages `ptr` and `cgo` are used internally to simplify and optimize the state management and the usage of C-allocated memory pointers.
+
+For some use cases, developers can consider using the SDK layers selectively. This is meaningful only if developers wish to manually write part of the low-level C details of the framework in their plugins, but still want to use some parts of the SDK. However, this is discouraged if not for advanced use cases only. Developers are encouraged to use the `sdk/plugins` to build Falcosecurity plugins, which is easier to use and will have less frequent breaking changes.
+
+Further details can be found in the documentation of each package: [`sdk`](https://pkg.go.dev/github.com/falcosecurity/plugin-sdk-go/pkg/sdk), [`sdk/symbols`](https://pkg.go.dev/github.com/falcosecurity/plugin-sdk-go/pkg/sdk/symbols), and [`sdk/plugins`](https://pkg.go.dev/github.com/falcosecurity/plugin-sdk-go/pkg/sdk/plugins).
+
+### Getting Started
+The SDK is built on top a set of minimal composable interfaces describing the behavior of plugins and plugin instances. As such, developing plugins is as easy as defining a struct type representing the plugin itself, ensuring that the mandatory interface methods are defined on it, and then registering it to the SDK.
+
+To use the Go SDK, all you need to import are the `sdk` and `sdk/plugins` packages. The first contains all the core types and definitions used across the rest of the SDK packages, whereas the latter contains built-in constructs to develop plugins. The subpackages `sdk/plugins/source` and `sdk/plugins/extractor` contain specialized definitions for source and extractor plugins respectively.
+
+The `dummy` Go plugin, documented in the [next sections](#example-go-plugin-dummy), is a simple example that helps understanding how to start writing Go plugins with this SDK.
+
+#### Defining Extractor Plugins
+In the Go SDK, an extractor Plugin is a type implementing the following interface:
+```go
+// sdk/plugins/extractor
+type Plugin interface {
+	...
+	Info() *sdk.Info
+	Init(config string) error
+	// Destroy()
```

Comment:
Are the commented out methods intended to be optional? If so, we should mention that in the code block too. And since the notion of "optional" methods on an interface doesn't really exist, maybe we should consider all of these mandatory and deal with the small additional cost of having the plugin author write do-nothing modules rather than having these optional methods not really be part of the interface at all.

Reply:
I can't entirely agree with that. I believe we shouldn't hide the fact the plugin author can implement several interfaces (it's not a single interface), and some of those are optional. Perhaps I would instead remove the comment from there and introduce a different wording for optional interfaces. 

IMHO, saying something like is perfectly acceptable and understandable for a Go developer: 

> An extractor plugin can optionally implement the `Destroyer` interface. If so, `Destroy()` will be called to release resources used by a plugin before it gets destroyed by the framework.

Reply:
I think there's a good value proposition in having some interfaces as optional, so I endorse the comment of @leogr. I think adding a subsection such as "Optional Interface" might be the way to go here.

Reply:
If you want to keep them optional, that's fine. I see the section on the optional interfaces. But can we also include them in the code blocks?

Reply:
looks great thx!